library/src/main/java/com/otaliastudios/zoom/ZoomEngine.kt

```diff
@@ -773,58 +806,171 @@ internal constructor(context: Context) : ViewTreeObserver.OnGlobalLayoutListener
+                                zoomTargetY = zoomTarget.y,
+                                allowOverScroll = true, allowOverPinch = true)
+                    }
+                    // return here because new state will be ANIMATING
                     return
                 }
+                setState(NONE)
+            } finally {
+                resetPinchListenerState()
             }
-            setState(NONE)
+        }
+
+        /**
+         * Resets the fields of this pinch gesture listener
+         * to prepare it for the next pinch gesture detection
+         * and remove any remaining data from the previous gesture.
+         */
+        private fun resetPinchListenerState() {
+            mInitialAbsFocusPoint.set(Float.NaN, Float.NaN)
+            mCurrentAbsFocusOffset.set(0F, 0F)
+        }
+
+        /**
+         * Calculate pivot point to use for zoom based on pan fixes to be applied
+         *
+         * @param fixPan the amount of pan to apply to get into a valid state (no overscroll)
+         * @return x-axis and y-axis view coordinates
+         */
+        private fun calculateZoomPivotPoint(fixPan: AbsolutePoint): PointF {
+            if (zoom <= 1F) {
+                // The zoom pivot point here should be based on the gravity that is used
+                // to initially transform the content.
+                // Currently this is always [View.Gravity.CENTER] as indicated by [mTransformationGravity]
+                // but this might be changed by the user.
+                return AbsolutePoint(-mContentRect.width() / 2F, -mContentRect.height() / 2F).toViewCoordinate()
+            }
+
+            val x = when {
+                fixPan.x > 0 -> mContainerWidth // content needs to be moved left, use the right border as target
+                fixPan.x < 0 -> 0F // content needs to move right, use the left border as target
+                else -> mContainerWidth / 2F // axis is not changed, use center as target
+            }
+
+            val y = when {
+                fixPan.y > 0 -> mContainerHeight // content needs to be moved up, use the bottom border as target
+                fixPan.y < 0 -> 0F // content needs to move down, use the top border as target
+                else -> mContainerHeight / 2F // axis is not changed, use center as target
+            }
+
+            return PointF(x, y)
         }
     }
 
+    /**
+     * @return the currently set upper boundary for maximum zoom value
+     */
+    @Zoom
+    private fun getMaxZoom(): Float = mMaxZoom.toZoom(mMaxZoomMode)
+
+    /**
+     * @return the currently set lower boundary for minimum zoom value
+     */
+    @Zoom
+    private fun getMinZoom(): Float = mMinZoom.toZoom(mMinZoomMode)
 
     private inner class FlingScrollListener : GestureDetector.SimpleOnGestureListener() {
 
         override fun onDown(e: MotionEvent): Boolean {
+            // cancel any animation that might still be running
+            setState(NONE)
```

Comment:
I see some issues with this: zoom in, then fling fast toward one end, then put your finger when the content is overscrolled. It does not always come back.

Might be easy to fix, but we can leave it for another PR I think.

Reply:
ok I will just revert this little change in this PR then.